feat: replace all occurences of twitter with X

bassgeta opened the pull request:
## Description

Reran codegen, replaced all occurences of `Twitter` with `X` in our app :writing_hand: 

## Testing

Let's check all occurences of twitter and see if they are replaced with X

1. The user menu should have a link to our X and the link should work
![image](https://github.com/user-attachments/assets/04ec34e1-e83d-49c7-87bc-42e7153da37a)
2. The colony social links should include X in the dropdown
![image](https://github.com/user-attachments/assets/78a2f51a-5b3e-4d18-9b88-de5b71f74884)
3. WHen you try to edit a colony, the X link should show up
![image](https://github.com/user-attachments/assets/9097119e-f9a4-4780-9b3a-3536c4066428)
4. Also in the social links modal
![image](https://github.com/user-attachments/assets/b6bb28a9-f9e9-4d98-88b9-87ad576d8eec)
![image](https://github.com/user-attachments/assets/67e75df1-2af4-4252-9b9c-ce3aa9499fad)
5. And the completed action view 
![image](https://github.com/user-attachments/assets/30b4fbdc-c6ac-4650-9402-e485658dcbd0)


## Diffs

**New stuff** ✨

* New stuff goes here

**Changes** 🏗

* Changes go here

**Deletions** ⚰️

* Deletions go here

## TODO

- [ ] Add TODOs

(Resolves | Contributes to) #31415

melyndav commented:
Hey @bassgeta - it also needs to be changed across the landing page and beta invite widgets as well - https://www.figma.com/design/j0VEfFryOkmjrVVq2PItso/Landing-page-%2B-Beta-Launch?node-id=17258-4949&t=z6sNdHr6eNcDSuAG-4

<img width="578" alt="image" src="https://github.com/user-attachments/assets/8ab1d1ef-9fe9-4438-bb51-f5ab4901a6d3" />

iamsamgibbs commented:
Just a note to say I opened the same issue months ago which has some extra requirements #3071

The logo in the modal pop up should use `text-gray-900` rather than the twitter blue

![image](https://github.com/user-attachments/assets/e6d7273b-65df-4401-a510-497a8f190afb)

I also had this note:

> `Twitter` is also used as a name for external links in the database and is used as a key to access the value in places in the app. It may be possible to leave this as `Twitter` in the database, and purely update it in the front end - but it may be worth updating these values in the database instead for the sake of consistency (and will make accessing these key value pairs easier).

I'm guessing running codegen accounts for this but we might need to have a migration script to change the existing URLs from twitter to x?

bassgeta commented:
> Just a note to say I opened the same issue months ago which has some extra requirements #3071
> 
> The logo in the modal pop up should use `text-gray-900` rather than the twitter blue
> 
> ![image](https://private-user-images.githubusercontent.com/34915414/403495981-e6d7273b-65df-4401-a510-497a8f190afb.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzcwMjUxNTgsIm5iZiI6MTczNzAyNDg1OCwicGF0aCI6Ii8zNDkxNTQxNC80MDM0OTU5ODEtZTZkNzI3M2ItNjVkZi00NDAxLWE1MTAtNDk3YThmMTkwYWZiLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAxMTYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMTE2VDEwNTQxOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTExYWI0MDlkODU4YzEzMjY3MzBhOTM0YzljMzg1YTc0YTE0NmIxOGMzOGQxMTUyOTM1OTIyOTAwNGMzNmQ4MzYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.htpPzxNSgCTrgfHJeUWB8qOoJW2q8aCWMZAQg7YnsLg)
> 
> I also had this note:
> 
> > `Twitter` is also used as a name for external links in the database and is used as a key to access the value in places in the app. It may be possible to leave this as `Twitter` in the database, and purely update it in the front end - but it may be worth updating these values in the database instead for the sake of consistency (and will make accessing these key value pairs easier).
> 
> I'm guessing running codegen accounts for this but we might need to have a migration script to change the existing URLs from twitter to x?

THanks for the link color hint, fixed it and rebased :sunglasses: 
And yeah, in the end I've just left the API value to be twitter, but just added an override for the `useSocialLinksTable` hook :shrug:

bassgeta commented:
> Hey @bassgeta - it also needs to be changed across the landing page and beta invite widgets as well - https://www.figma.com/design/j0VEfFryOkmjrVVq2PItso/Landing-page-%2B-Beta-Launch?node-id=17258-4949&t=z6sNdHr6eNcDSuAG-4
> <img alt="image" width="578" src="https://private-user-images.githubusercontent.com/82127904/403488571-8ab1d1ef-9fe9-4438-bb51-f5ab4901a6d3.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzcwMjUxNTgsIm5iZiI6MTczNzAyNDg1OCwicGF0aCI6Ii84MjEyNzkwNC80MDM0ODg1NzEtOGFiMWQxZWYtOWZlOS00NDM4LWJiNTEtZjVhYjQ5MDFhNmQzLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAxMTYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMTE2VDEwNTQxOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWUwMThmNDZlZWE5YmY5MTEzMjkwOGYxYjk3MDEwYjY2OWYxMTdkZmQ0MzNmMTc4NWUxYTA3ZDVlODVmNmZkNzMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.0hYVNxSw20SHtQI14mEcTEYR2YX2BuAWxA21HizDJ2I">

![image](https://github.com/user-attachments/assets/bf7c5476-f696-4e04-a6ac-f038ad3fc2d7)
:saluting_face:

iamsamgibbs reviewed (approved):
Nice one, this all looks good to me! Goodbye birdy.

![Screenshot 2025-01-16 at 11 02 50](https://github.com/user-attachments/assets/2a8ba6d1-282f-4ace-96f5-e5c945280acb)
![Screenshot 2025-01-16 at 11 03 37](https://github.com/user-attachments/assets/6c6953f1-1049-4fcc-835f-64367f261c3a)
![Screenshot 2025-01-16 at 11 03 56](https://github.com/user-attachments/assets/04e84533-fb17-4971-b947-1fb6de54ce53)
![Screenshot 2025-01-16 at 11 04 31](https://github.com/user-attachments/assets/17280386-3d9d-4808-9612-9c4db1f5ca0f)
![Screenshot 2025-01-16 at 11 04 34](https://github.com/user-attachments/assets/e32cce3d-6285-4780-882f-b73d03d80c3b)

Unrelated to this issue, but we should maybe think about changing the way this works and prepending the site URLs ourselves as you could put any URL you like with the X logo and could potentially be used maliciously.

![Screenshot 2025-01-16 at 11 06 47](https://github.com/user-attachments/assets/67d4fd6e-391c-4028-8627-6e0bfd7706ff)

rdig reviewed (approved):
Very nice! Now everything is renamed to the "everything app" :)

![Screenshot from 2025-01-16 14-02-47](https://github.com/user-attachments/assets/a6fa28f7-3a27-4ec8-8faf-d253e333c796)
![Screenshot from 2025-01-16 14-03-03](https://github.com/user-attachments/assets/57aa80c4-c375-4568-ab68-08c7938379dd)

mmioana reviewed (approved):
Twitter-who? Nice work @bassgeta now all labels and icons point to `X` 😎 

![Screenshot 2025-01-17 at 13 51 18](https://github.com/user-attachments/assets/0221ba12-90c9-48b9-a972-60bd3431a60e)
![Screenshot 2025-01-17 at 13 51 36](https://github.com/user-attachments/assets/3324be30-5c37-46b2-86f7-ab720bd72853)
![Screenshot 2025-01-17 at 13 52 03](https://github.com/user-attachments/assets/5be82a30-7859-42af-b8c3-6088decc6b0a)
![Screenshot 2025-01-17 at 13 52 12](https://github.com/user-attachments/assets/8aa279a5-765c-487d-b851-fd68b30dea0e)
![Screenshot 2025-01-17 at 13 52 25](https://github.com/user-attachments/assets/ae07c46b-a402-4e39-9ae7-9739eed9d81a)
![Screenshot 2025-01-17 at 13 53 30](https://github.com/user-attachments/assets/7a700ccb-be69-41b2-bdca-f37b60ff7206)
![Screenshot 2025-01-17 at 13 53 57](https://github.com/user-attachments/assets/aca8a00c-0d69-4516-89ac-8d34806b92ea)
